Add documentation on Azure ADLS support (#2506)
duckdb/duckdb-web: docs/extensions/azure.md

## Usage

> Once the [authentication](#authentication) is set up, you Azure storages can as follows:

Reviewer: ```suggestion
Once the [authentication](#authentication) is set up, you can query Azure storage as follows:
```

| `azure_http_proxy`| Proxy to use when login & performing request to azure. | `STRING` | `HTTP_PROXY` environment variable (if set). |
| `azure_proxy_user_name`| Http proxy username if needed. | `STRING` | - |
| `azure_proxy_password`| Http proxy password if needed. | `STRING` | - |

## Additional information

> ### Why implement the two storage types

Reviewer: ```suggestion
### Difference between ADLS and Blob storage
```

> Even if the DataLake storage also expose the Blob endpoint adding the support of the DataLake allow users to add some predicate in there paths.

Reviewer: ```suggestion
Even though ADLS implements similar functionality as the Blob storage, there are some important performance benefits to using the ADLS endpoints for globbing, especially when using (complex) glob patterns. 
```

> For example the main difference in behavior for the following use case:

Reviewer: ```suggestion
To demonstrate we will give an example of how the a glob is performed internally using respectively the Glob and ADLS endpoints.
```

```sql
SELECT count(*)
FROM 'abfss://root/l_receipmonth=1997-*/l_shipmode=SHIP/*.csv';
```

Will perform the following steps:

* List all directories in `root/`
  * root/l_receipmonth=1997-10
  * root/l_receipmonth=1997-11
  * root/l_receipmonth=1997-12
* Filter and list subdirectories: `root/l_receipmonth=1997-10`, `root/l_receipmonth=1997-11`, `root/l_receipmonth=1997-12`
  * root/l_receipmonth=1997-10/l_shipmode=SHIP
  * root/l_receipmonth=1997-10/l_shipmode=AIR
  * root/l_receipmonth=1997-10/l_shipmode=TRUCK
  * root/l_receipmonth=1997-11/l_shipmode=SHIP
  * root/l_receipmonth=1997-11/l_shipmode=AIR
  * root/l_receipmonth=1997-11/l_shipmode=TRUCK
  * root/l_receipmonth=1997-12/l_shipmode=SHIP
  * root/l_receipmonth=1997-12/l_shipmode=AIR
  * root/l_receipmonth=1997-12/l_shipmode=TRUCK
* Filter and list subdirectories: `root/l_receipmonth=1997-10/l_shipmode=SHIP`, `root/l_receipmonth=1997-11/l_shipmode=SHIP`, `root/l_receipmonth=1997-12/l_shipmode=SHIP`
  * root/l_receipmonth=1997-10/l_shipmode=SHIP/data_0.csv
  * root/l_receipmonth=1997-11/l_shipmode=SHIP/data_0.csv
  * root/l_receipmonth=1997-12/l_shipmode=SHIP/data_0.csv

> As you can see because blob do not support the notion of directories the filter can only be performed after the listing whereas the datalake endpoint will list recursively directory by directory. In the example it does not look that different but with more partitions or more files under the partitions that can make a lot of differences.

Reviewer: ```suggestion
As you can see because the Blob endpoint does not support the notion of directories, the filter can only be performed after the listing, whereas the ADLS endpoint will list files recursively. Especially with higher partition/directory counts, the performance difference can be very significant.
```
